src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs

```diff
@@ -155,7 +157,19 @@ private void WriteDepsFile(string depsFilePath)
                 RuntimeFrameworks,
                 IsSelfContained);
 
-            var builder = new DependencyContextBuilder(mainProject, projectContext, IncludeRuntimeFileVersions);
+            // Generate the RID-fallback for self-contained builds.
+            //
+            // In order to support loading components with RID-specific assets, the AssemblyDependencyResolver requires a RID fallback graph.
+            // The component itself should not carry the RID fallback graph with it (it would need to have the graph of all the RIDs there are and it would need to be updated with every addition).
+            // For framework dependent apps, the RID fallback graph comes from the core framework Microsoft.NETCore.App, so there is no need to write it into the app.
+            // If self-contained apps, the (applicable subset of) RID fallback graph needs to be written to the deps.json manifest.
+            //
+            // If a RID-graph is provided to the DependencyContextBuilder, it generates a RID-fallback graph wrt the current RuntimeIdentifier.
```

Comment:
What is "current" in this context. Does this work if cross-publishing to a different RID than build host?

Reply:
Its the target RID for which the app is being built, I'll fix the comment wording. Thanks.

Reply:
Ah, I see the RID on project context. I was wondering where it would get it from.